commit 4a659cb2f0fd7a1d8ddebc687e922cc10bff719b
Author: Fuyun Liang <liangfuyun1@huawei.com>
Date:   Fri Jul 6 11:27:59 2018 +0100

    net: hns3: Fix for mailbox message truncated problem
    
    [ Upstream commit ead5bd4d35c0a14d5ce1474177718c678dff5205 ]
    
    The payload of mailbox message is 16 byte and the value of
    HCLGE_MBX_MAX_ARQ_MSG_SIZE is 8. A message truncated problem will
    happen when mailbox message is converted to ARQ message. This patch
    replaces HCLGE_MBX_MAX_ARQ_MSG_SIZE with the size of ARQ message in
    hclgevf_mbx_handler to fix this problem.
    
    Fixes: b11a0bb231f3 ("net: hns3: Add mailbox support to VF driver")
    Signed-off-by: Fuyun Liang <liangfuyun1@huawei.com>
    Signed-off-by: Peng Li <lipeng321@huawei.com>
    Signed-off-by: Salil Mehta <salil.mehta@huawei.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/hisilicon/hns3/hns3vf/hclgevf_mbx.c b/drivers/net/ethernet/hisilicon/hns3/hns3vf/hclgevf_mbx.c
index b598c06af8e0..cd246f906150 100644
--- a/drivers/net/ethernet/hisilicon/hns3/hns3vf/hclgevf_mbx.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3vf/hclgevf_mbx.c
@@ -208,7 +208,8 @@ void hclgevf_mbx_handler(struct hclgevf_dev *hdev)
 
 			/* tail the async message in arq */
 			msg_q = hdev->arq.msg_q[hdev->arq.tail];
-			memcpy(&msg_q[0], req->msg, HCLGE_MBX_MAX_ARQ_MSG_SIZE);
+			memcpy(&msg_q[0], req->msg,
+			       HCLGE_MBX_MAX_ARQ_MSG_SIZE * sizeof(u16));
 			hclge_mbx_tail_ptr_move_arq(hdev->arq);
 			hdev->arq.count++;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1095_4a659cb2f0fd7a1d8ddebc687e922cc10bff719b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1095_4a659cb2f0fd7a1d8ddebc687e922cc10bff719b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
